### PR TITLE
[fix] GH App Auth Validator missing db

### PR DESCRIPTION
**Pull request by pjlast**
The validate external service config is panicking on GitHub App connections because the `db` passed to the validator is `nil`.

This PR ensures that a DB is passed to the validator.

## Test plan

Tests passing.

### CODE REVIEW

**Comment by mrnugget** [comment]
> The validate external service config is panicking on GitHub App connections because the `db` passed to the validator is `nil`.

Wait, how did this not show up before locally?

File: internal/database/external_services.go [hunk 1]
```diff
@@ -313,13 +313,13 @@ type ValidateExternalServiceConfigOptions struct {
 	AuthProviders []schema.AuthProviders
 }
 
-type ValidateExternalServiceConfigFunc = func(ctx context.Context, e ExternalServiceStore, opt ValidateExternalServiceConfigOptions) (normalized []byte, err error)
+type ValidateExternalServiceConfigFunc = func(ctx context.Context, db DB, opt ValidateExternalServiceConfigOptions) (normalized []byte, err error)
```
Comment (sashaostrikov) [inline internal/database/external_services.go:316]: Interesting and a bit strange that `e ExternalServiceStore` wasn't used at all.

File: internal/database/external_services.go [hunk 2]
```diff
@@ -313,13 +313,13 @@ type ValidateExternalServiceConfigOptions struct {
...
 var ValidateExternalServiceConfig = MakeValidateExternalServiceConfigFunc(nil, nil, nil, nil, nil)
 
-func MakeValidateExternalServiceConfigFunc(gitHubValidators []func(*types.GitHubConnection) error, gitLabValidators []func(*schema.GitLabConnection, []schema.AuthProviders) error, bitbucketServerValidators []func(*schema.BitbucketServerConnection) error, perforceValidators []func(*schema.PerforceConnection) error, azureDevOpsValidators []func(connection *schema.AzureDevOpsConnection) error) ValidateExternalServiceConfigFunc {
-	return func(ctx context.Context, e ExternalServiceStore, opt ValidateExternalServiceConfigOptions) (normalized []byte, err error) {
+func MakeValidateExternalServiceConfigFunc(gitHubValidators []func(DB, *types.GitHubConnection) error, gitLabValidators []func(*schema.GitLabConnection, []schema.AuthProviders) error, bitbucketServerValidators []func(*schema.BitbucketServerConnection) error, perforceValidators []func(*schema.PerforceConnection) error, azureDevOpsValidators []func(connection *schema.AzureDevOpsConnection) error) ValidateExternalServiceConfigFunc {
```
Comment (sashaostrikov) [inline internal/database/external_services.go:321]: Let's split this line, it's very long :)

**Review by sashaostrikov — COMMENTED** [review]
I wish such kinds of things are caught by unit tests. Do you think it's possible to write one for such a case?

File: internal/authz/providers/github/authz_test.go [hunk 3]
```diff
@@ -218,3 +249,53 @@ func TestNewAuthzProviders(t *testing.T) {
...
+          GroupsCacheTTL: 72,
+        },
+        GitHubAppDetails: &schema.GitHubAppDetails{
+          AppID: 1,
+          BaseURL: schema.DefaultGitHubURL,
+          InstallationID: 1,
+        },
+      },
+    },
+  }
```
Comment (sashaostrikov) [inline internal/authz/providers/github/authz_test.go:289]: Thanks for the tests! Do you thing it might be worth adding a negative case too?

**Review by sashaostrikov — APPROVED** [review]
LGTM! Nice refactoring also 👍🏻

File: internal/authz/providers/github/authz_test.go (hunk 3 above)
Comment (kopancek) [reply internal/authz/providers/github/authz_test.go:289]: Agreed, it would be good to test the case when `dbm == nil`

**Comment by pjlast** [comment]
But `dbm` shouldn't ever be `nil` . That's the issue.

It slipped through previously because `ValidateAuthz` had zero tests, and when the functionality was changed to depend on the `db`, there was no test that caught that fact.

So I think this should be fine.

There isn't a good argument for `db` ever being `nil` imo

**Comment by sashaostrikov** [comment]
@pjlast by "negative" case I meant when the validation doesn't pass and we receive an error, sorry for ambiguity :)

**Comment by mrnugget** [comment]
> It slipped through previously because `ValidateAuthz` had zero tests, and when the functionality was changed to depend on the `db`, there was no test that caught that fact.

It should've shown up in manual testing though right?

**Comment by pjlast** [comment]
@mrnugget there were a bunch of changes here recently and I think it the manual case just wasn't hit. Like:

1. Go to GitHub connection. Click test connection. Works
2. Go to GitHub App connection, click test connection. Works
3. Create new GitHub connection/update GitHub connection. Works

And then the new/update case for GHApp just wasn't done

**Comment by mrnugget** [comment]
> And then the new/update case for GHApp just wasn't done

Okay, got it!

**Pull request merged**